### ISSUE TITLE
False Positive CS8604 for Nullable<T>.ToString()

### ISSUE THREAD

**Issue by realbart** [dotnet/roslyn#68424]
This code should not produce a CS8604 warning:

```
        int? value = 1;
        if (value is not null)
        {
            string stringValue = value.ToString();
        }
```

It's true that Nullable<int>.ToString() returns a nullable string, but it has only a null value if HasValue is false.

**Comment by RikkiGibson** [comment]
It looks like this is about the nullable annotation of `System.Nullable<T>.ToString`, which is more of a dotnet/runtime issue.

I expect that the nullable annotation is intentional as the underlying T is allowed to return null from its `ToString`. We unfortunately don't support making the behavior depend on the ToString of the underlying type, for example.
